### PR TITLE
Issue 3496 - Select media from wp media library - Part Two

### PR DESCRIPTION
**Pull request by SergioEstevao**
Second step PR to address #3496.

The code now allows to sync the media library from the server.

To test it:

 * Create a new post and insert some images/videos.
 * Select the Media Library option on the media picker
 * If it's the first time you access it, it should start synchronising the media library from the server
 * After a while you should start to see some of media thumbnails showing.
 * Test by selecting images and videos.

Needs Review: @bummytime

### CODE REVIEW

**Comment by bummytime** [comment]
@SergioEstevao This is really looking good. The media lib sync and picker seem to be working well. I did come across one issue with videos however. Here is a video:

https://cloudup.com/cY3s0EMCayy

In short, inserting videos from the media lib does not work. You are able to insert the video into the post (and you can even play it), but after you save the post and re-open it, the video is missing. FWIW I did test uploading local videos and that still worked.

**Comment by bummytime** [comment]
@SergioEstevao The video issue I described above seems to be only a problem on .COM. .ORG sites work — I can insert and save media lib videos without a problem.

File: WordPress/Classes/Services/MediaService.m [hunk 1]
```diff
@@ -332,6 +442,37 @@ - (NSString *)mimeTypeForFilename:(NSString *)filename
     return remote;
 }
 
+- (void)mergeMedia:(NSArray *)media
+           forBlog:(Blog *)blog
+ completionHandler:(void (^)(void))completion
+{
```
Comment (bummytime) [inline WordPress/Classes/Services/MediaService.m]: Could you add some assertions here for the params?

File: WordPress/Classes/ViewRelated/Media/MediaLibraryPickerDataSource.m [hunk 2]
```diff
@@ -309,6 +317,14 @@ - (NSTimeInterval)duration
     if (self.mediaType != MediaTypeVideo){
         return 0;
     }
+    if (self.length != nil && [self.length doubleValue] > 0){
```
Comment (bummytime) [inline WordPress/Classes/ViewRelated/Media/MediaLibraryPickerDataSource.m]: Style nit: space after `)` and `{`

File: WordPress/Classes/ViewRelated/Media/MediaLibraryPickerDataSource.m [hunk 3]
```diff
@@ -309,6 +317,14 @@ - (NSTimeInterval)duration
...
+        return [self.length doubleValue];
+    }
+    
+    if (self.absoluteLocalURL == nil ||
+        ![[NSFileManager defaultManager] fileExistsAtPath:self.absoluteLocalURL isDirectory:nil]){
```
Comment (bummytime) [inline WordPress/Classes/ViewRelated/Media/MediaLibraryPickerDataSource.m]: Style nit: space after `)` and `{`

**Comment by SergioEstevao** [comment]
@bummytime Good catch regarding the videos in WP.com. I fixed that and the other code layout issues. Ready for another round.

**Comment by bummytime** [comment]
@SergioEstevao the fixed worked well. I did uncover another issue on non-jp .org sites. If you upload a video from the device, a `<video>` tag is inserted into the post. If you insert a video from wp-admin, a shortcode is used. 

napkin 9 06-22-15 10 41 11 am

I am not sure if this is an issue really...but in **Chrome** I did notice this in the visual editor:

napkin 8 06-22-15 10 36 16 am

The shortcode movie appears and the `<video>` tag movie is a small dot. Note, it doesn't seem to affect the published post from a reader's perspective. We can probably make a separate issue for this since it is out of scope of this PR.

**Comment by bummytime** [comment]
If we could make a separate issue for the thing I just noted above, I am giving this a :shipit:

**Comment by SergioEstevao** [comment]
Thanks, @bummytime the issue that you found was already present in the app so I opened a separate issue https://github.com/wordpress-mobile/WordPress-iOS/issues/3911 to track it.

**Pull request merged**